### ISSUE TITLE
Providing a default value for `selection` parameter causes view to be unscrollable

### ISSUE THREAD

**Issue by bstillitano** [AllanJuenemann/CalendarView#3]
Providing anything except `nil` as the default value for a `selection` binding, will cause the view to be non scrollable until the user taps another date.

**Comment by AllanJuenemann** [comment]
@bstillitano Sorry for the slow response.

I tried to reproduce the issue you described. See below. But all versions work fine (tested with version [2.1.0](https://github.com/AllanJuenemann/CalendarView/releases/tag/2.1.0) on iOS 17.5)

```swift
struct ContentView: View {
    @State private var selection: DateComponents?
    
    var body: some View {
        CalendarView(selection: $selection)
    }
}
```

```swift
struct ContentView: View {
    @State private var selection: DateComponents? = DateComponents(year: 2024, month: 8, day: 16)
    
    var body: some View {
        CalendarView(selection: $selection)
    }
}
```

```swift
struct ContentView: View {
    @State private var selection: DateComponents? 
    
    var body: some View {
        CalendarView(selection: $selection)
            .onAppear {
                selection = DateComponents(year: 2024, month: 8, day: 16)
            }
    }
}
```

Would you be able to share a code sample with me that causes the scrolling issues you're experiencing?

**Comment by lishuhao** [comment]
I have the same problem
This example can reproduce the issue:

```swift
struct ContentView: View {
    @State private var visibleComponents: DateComponents = Calendar.current.dateComponents([.year, .month], from: Date())
    @State private var selection: DateComponents? = DateComponents(year: 2024, month: 8, day: 16)
    
    var body: some View {
        CalendarView(visibleDateComponents: $visibleComponents, selection: $selection)
    }
}
```

**Comment by AllanJuenemann** [comment]
Thank you @lishuhao! This demonstrates the issue nicely. I'll get this fixed.

**Comment by AllanJuenemann** [comment]
@lishuhao @bstillitano This issue should now be fixed in version [2.1.2](https://github.com/AllanJuenemann/CalendarView/releases/tag/2.1.2).
